### PR TITLE
Api reference: Deprecated disableRedirect param and direct-upload kvs endpoint

### PR DESCRIPTION
**Pull request by drobnikj**
It is not aligned with the current state of API regarding the decompression of record. We decompress just records requested with disableRedirect param. We plan to do it, I will merge this after.

### CODE REVIEW

File: docs/api_v2/api_v2_reference.apib [hunk 1]
```diff
@@ -2309,12 +2307,12 @@ This endpoint is paginated using `exclusiveStartKey` and `limit` parameters - se
 
 Gets a value stored in the key-value store under a specific key.
 
-If the request defines the `Accept-Encoding: gzip` HTTP header then the response will be gzipped.
+The response body has the same `Content-Encoding` header as it was set in [Put record](#reference/key-value-stores/key-collection/put-record).
+If the request does not define the `Accept-Encoding` HTTP header with the right encoding, the record will be decompressed.
+The most HTTP client supports decompression by default. After using the HTTP client with decompression support, the `Accept-Encoding` header is set by the client and body is decompressed automatically.
```
Comment (m-murasovs) [inline docs/api_v2/api_v2_reference.apib:2312]: Here, do you mean "Most HTTP clients support... "?

File: docs/api_v2/api_v2_reference.apib [hunk 2]
```diff
@@ -2332,13 +2330,15 @@ If the request defines the `Accept-Encoding: gzip` HTTP header then the response
...
 
 Stores a value under a specific key to the key-value store.
-The value is passed as the PUT payload and it is stored with a MIME content type defined by the `Content-Type` request header.
+The value is passed as the PUT payload and it is stored with a MIME content type defined by the `Content-Type` header and with encoding defined by the `Content-Encoding` header.
 
-**IMPORTANT:** The limit of the request payload is 9 MB. If you want to upload a larger record
-or speed up your upload, use the [Direct upload URL](#reference/key-value-stores/direct-upload-url) endpoint instead.
+To save bandwidth, storage, and speed up your upload, send the request payload compressed with Gzip compression and
+add the `Content-Encoding: gzip` header. It is possible to set up another compression type with `Content-Encoding` request header.
+There is a list of supported `Content-Encoding` types.
```
Comment (m-murasovs) [inline docs/api_v2/api_v2_reference.apib:2337]: Please change to "Below is a list..."

**Review by m-murasovs — APPROVED** [review]
Looks good, just minor grammar stuff :)

**Review by mtrunkat — APPROVED** [review]

**Review by m-murasovs — CHANGES_REQUESTED** [review]
Sorry, approved before by accident - a couple minor things :D

**Review by m-murasovs — APPROVED** [review]

**Pull request merged**